### PR TITLE
[Monitoring] Add new cluster privilege now necessary for the stack monitoring ui

### PR DESCRIPTION
**Pull request by chrisronline**
Resolves https://github.com/elastic/kibana/issues/47859

We added a new call in the Stack Monitoring UI that requires this cluster privilege.

To test, create a user with the minimum permissions (roles=[`kibana_user`, `monitoring_user`]) and ensure you can access the Stack Monitoring UI in Kibana

### CODE REVIEW

**Comment by pickypg** [comment]
For anyone that runs into this issue, there is a workaround for those running 7.4.0 in the short term by creating a temporary, custom `role`:

```http
PUT /_security/role/fix_monitoring_user
{
  "cluster": [
    "cluster:monitor/remote/info"
  ]
}
```

then adding it to your monitoring `user` (the key is the extra `role`):

```http
PUT /_security/user/my_monitoring_user
{
  "username": "my_monitoring_user",
  "roles": [
    "kibana_user",
    "monitoring_user",
    "fix_monitoring_user"
  ],
  "enabled": true,
  "full_name": "My Monitoring User",
  "email": "my-user@company.test"
}
```

Note: this implies a user update, since there's no password.

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
Based on my recent experience in opening a PR against these files, I think this may need to be reviewed by @elastic/es-security

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java [hunk 1]
```diff
@@ -54,7 +54,7 @@
                         null, null,
                         MetadataUtils.DEFAULT_RESERVED_METADATA, null))
                 .put("monitoring_user", new RoleDescriptor("monitoring_user",
-                        new String[] { "cluster:monitor/main", "cluster:monitor/xpack/info" },
+                        new String[] { "cluster:monitor/main", "cluster:monitor/xpack/info", "cluster:monitor/remote/info" },
```
Comment (igoristic) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java]: More of an observation, maybe we can use the static member `NAME` from action `RemoteInfoAction.NAME`. And, we can also do that for `cluster:monitor/main` and `cluster:monitor/remote/info`. Don't know if they're meant to be used like constants though

**Review by igoristic — APPROVED** [review]
Works as expected 🥇

**Review by pickypg — CHANGES_REQUESTED** [review]
This needs to update tests in

https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java#L433, which conveniently will provide a chance to use @igoristic's recommendation in the test.

**Review by pickypg — APPROVED** [review]
LGTM.

I'm not sure how the ES team feels about mixing the constants with the static references, so I'll leave that for them to make a final ruling on.

**Comment by chrisronline** [comment]
[There is precedent for doing this right now](https://github.com/elastic/elasticsearch/pull/47871/files#diff-b948efc6931697715ce5dd7145eb7decR115)

**Review by tvernum — APPROVED** [review]

**Comment by chrisronline** [comment]
Backport:

7.4: https://github.com/elastic/elasticsearch/commit/157152757545ebe2112d8750a397544f619f7166
7.x: https://github.com/elastic/elasticsearch/commit/c57191b16398d78fdf114e3159f288ce722adec8

**Pull request merged**